be/src/storage/column_in_predicate.cpp

```diff
@@ -365,6 +367,93 @@ class BinaryColumnInPredicate : public ColumnPredicate {
     ItemHashSet<Slice> _slices;
 };
 
+class DictionaryCodeInPredicate : public ColumnPredicate {
+private:
+    enum LogicOp { ASSIGN, AND, OR };
+
+public:
+    DictionaryCodeInPredicate(const TypeInfoPtr& type_info, ColumnId id, const std::vector<int32_t>& operands,
+                              size_t size)
+            : ColumnPredicate(type_info, id), _bit_mask(size) {
+        for (auto item : operands) {
+            DCHECK(item < size);
+            _bit_mask[item] = 1;
+        }
+    }
+
+    ~DictionaryCodeInPredicate() override = default;
+
+    template <LogicOp Op>
+    inline void t_evaluate(const Column* column, uint8_t* sel, uint16_t from, uint16_t to) const {
+        const Int32Column* dict_code_column;
+        if (column->is_nullable()) {
+            // This is NullableColumn, get its data_column
+            dict_code_column =
+                    down_cast<const Int32Column*>(down_cast<const NullableColumn*>(column)->data_column().get());
+        } else {
+            dict_code_column = down_cast<const Int32Column*>(column);
+        }
+        std::vector<int32_t> data = dict_code_column->get_data();
+        Filter filter(to - from, 1);
+
+        if (column->has_null()) {
+            const NullColumn* null_column = down_cast<const NullableColumn*>(column)->null_column().get();
+            auto null_data = null_column->get_data();
+            for (auto i = from; i < to; i++) {
+                filter[i - from] = !null_data[i];
+            }
+        }
+
+        for (auto i = from; i < to; i++) {
+            auto index = data[i] >= _bit_mask.size() ? 0 : data[i];
+            filter[i - from] &= _bit_mask[index];
+        }
+
+        for (auto i = from; i < to; i++) {
```

Comment:
Is it better to merge this for-loop with the one in LOC:407

Reply:
agree, will do it.